Use eccentricity to find graph centres

pradkrish opened the pull request:
Partially fixes #2027

szhorvat commented on src/paths/distances.c:
```diff
@@ -212,3 +212,34 @@ int igraph_radius(const igraph_t *graph, igraph_real_t *radius,
+
+    igraph_eccentricity(graph, &ecc, vids, mode);
+    IGRAPH_CHECK(igraph_vit_create(graph, vids, &vit));
+    if (!IGRAPH_VIT_END(vit)) {
```
This loop is unnecessary. We only need to find the vertices with maximum eccentricity, so we only need to loop over the eccentricity vector.

pradkrish replied:
The idea of adding the condition `if (!IGRAPH_VIT_END(vit))` is to check for Null graph. If this is not correct, is there another way of doing it?

szhorvat replied:
Good point that we must check for the null graph, otherwise `vector_min` fails. Do this by checking the vertex count.

szhorvat reviewed (commented):
I added a few preliminary comments, but I think it will be easier for you to open a new PR against `develop` instead of rebasing this one. It's up to you which way to proceed though.

szhorvat commented:
FYI the test failures with MinGW are not related to your code, and have just been worked around `master` and `develop`. After you rebase this on `develop` (or open a new PR against `develop`), this annoyance will be gone.

pradkrish commented on src/paths/distances.c:
```diff
@@ -212,3 +212,34 @@ int igraph_radius(const igraph_t *graph, igraph_real_t *radius,
+    igraph_vit_t vit;
+    igraph_vs_t vids = igraph_vss_all();
+
+    igraph_vector_init(&ecc, 0);
```
Ack

pradkrish commented on src/paths/distances.c:
```diff
@@ -212,3 +212,34 @@ int igraph_radius(const igraph_t *graph, igraph_real_t *radius,
+    return (a > b) ? b : a;
+}
+
+int igraph_center(const igraph_t *graph, igraph_vector_t *res,
```
Ack

pradkrish commented:
> FYI the test failures with MinGW are not related to your code, and have just been worked around `master` and `develop`. After you rebase this on `develop` (or open a new PR against `develop`), this annoyance will be gone.

Yeah, it was an oversight on my part, I thought I was on develop branch. Also, it looks like the unit tests I have added is not building as part of the test suite.. I have added it to `tests/CMakeList.txt`. Am I missing something here?

pradkrish commented:
That doesn't look good at all, best to abandon this and open a new PR.
